coalib/core/CircularDependencyError.py

```diff
@@ -3,7 +3,10 @@ class CircularDependencyError(Exception):
     An error identifying a circular dependency.
     """
 
-    def __init__(self, node=None):
+    prefix = 'Circular dependency detected'
```

Comment:
Is `prefix` intended to be changeable? If not, there is no need to store it in an attribute.

And in any case, I would prepend that prefix in `.__str__` instead of in every `.for_...` method